src/runtime/virtcontainers/pkg/oci/utils.go

```diff
@@ -551,57 +537,36 @@ func addHypervisorPathOverrides(ocispec specs.Spec, config *vc.SandboxConfig, ru
 }
 
 func addHypervisorMemoryOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfig, runtime RuntimeConfig) error {
-	if value, ok := ocispec.Annotations[vcAnnotations.DefaultMemory]; ok {
-		memorySz, err := strconv.ParseUint(value, 10, 32)
-		if err != nil {
-			return fmt.Errorf("Error encountered parsing annotation for default_memory: %v, please specify positive numeric value greater than 8", err)
-		}
 
+	if err := parseAnnotationUint32Configuration(ocispec, vcAnnotations.DefaultMemory, &sbConfig.HypervisorConfig.MemorySize, func(memorySz uint32) error {
 		if memorySz < vc.MinHypervisorMemory {
 			return fmt.Errorf("Memory specified in annotation %s is less than minimum required %d, please specify a larger value", vcAnnotations.DefaultMemory, vc.MinHypervisorMemory)
 		}
 
-		sbConfig.HypervisorConfig.MemorySize = uint32(memorySz)
+		return nil
```

Comment:
@liubin, this is rather confusing to read, and I do believe we'd be better having the function defined somewhere else and then used here.

At the first glance (and that was myself Yesterday, tired, looking at this code) I had the impression you were doing an early return here.  And I'm pretty sure I'll think the same whenever I'll have to debug it, for one reason or another.

So, instead of just implementing the function as it's now, can we have it defined out of call to parseAnnotationUint32Configuration() and then just have it passed to parseAnnotationUint32Configuration()?